ci/jenkins/stop-jobs.sh

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+
+function echoerr {
+    >&2 echo "$@"
+}
+
+PULL_REQUEST_ID=""
+JENKINS_URL=""
+DEFAULT_WORKDIR=$HOME
+
+_usage="Usage: $0 [--pull-request] [--jenkins]
+
+        --pull-request           Unique Pull Request id, to verify the stale jobs that needs to be aborted.
+        --jenkins                Jenkins url to fetch the information about jobs."
+
+function print_usage {
+    echoerr "$_usage"
+}
+
+function print_help {
+    echoerr "Try '$0 --help' for more information."
+}
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    --pull-request)
+    PULL_REQUEST_ID="$2"
+    shift 2
+    ;;
+    --jenkins)
+    JENKINS_URL="$2"
+    shift 2
+    ;;
+    -h|--help)
+    print_usage
+    exit 0
+    ;;
+    *) 
+    echoerr "Unknown option $1"
+    exit 1
+    ;;
+esac
+done
+
+function abort_queue_jobs {
+    echo "====== checking for capv stale waiting jobs on PR ${PULL_REQUEST_ID} ======"
+    capv_job_name=$1
+    length_of_queue=$(curl -s -XGET "${JENKINS_URL}queue/api/json" | jq '[.[]][2]' | jq length)
+
+    for (( j=0; j<$length_of_queue; j++ )); do
+        job_name=$(curl -s -XGET "${JENKINS_URL}queue/api/json" | jq "[.[]][2][$j].task.name" | tr -d '"') 
+        if [[ "$job_name" == "$capv_job_name" ]]; then
+            # fetch the waiting job id and its pr id.
+            job_id=$(curl -s -XGET "${JENKINS_URL}queue/api/json" | jq "[.[]][2][$j].id")
+            waiting_job_pr_id=$(curl -s -XGET "${JENKINS_URL}queue/api/json" | jq "[.[]][2][$j].actions[1].parameters[9].value" | tr -d '"')
+
+            if [[ "$PULL_REQUEST_ID" == "$waiting_job_pr_id" ]]; then
+                echo "====== abort waiting job $capv_job_name on PR ${PULL_REQUEST_ID} ======"
+                curl -XPOST "${JENKINS_URL}/queue/cancelItem?id=$job_id" -u admin:$JENKINS_API_TOKEN
+                break
+            fi
+        fi
+    done
+}
+
+function abort_running_jobs {
+    echo "====== checking for capv stale running jobs on PR ${PULL_REQUEST_ID} ======"
+    capv_job_name=$1
+    i=$2
+    job_color=$(curl -s -XGET "${JENKINS_URL}label/antrea-test-node/api/json" | jq "[.[]][10][$i].color" | tr -d '"')
+
+    if [[ ${job_color: -5} == "anime" ]]; then
```

Comment:
Can you add a comment to explain what is job color? or add an official doc link for it.

Reply:
job_color is the status of the jobs. for example if job is in running state then `job color ` value will be `red_anime` or `blue_anime`, here "anime" means job is in running state.
red_anime: last failed job is in running state.
blue_anime: last successful job is in running state.

curl output:
```
> curl -s -XGET "http://10.176.27.169:8080/label/antrea-test-node/api/json" | jq "[.[]][10][$i].color" | tr -d '"'

disabled
blue
disabled
blue
blue
red
notbuilt
blue_anime
red
blue
```

you can check color field in the curl:
```
> curl -s -XGET "http://10.176.27.169:8080/label/antrea-test-node/api/json" | jq "[.[]][10][$i]"
{
  "_class": "hudson.model.FreeStyleProject",
  "name": "antrea-all-conformance-for-pull-request",
  "url": "http://10.176.27.169:8080/job/antrea-all-conformance-for-pull-request/",
  "color": "disabled"
}
{
  "_class": "hudson.model.FreeStyleProject",
  "name": "antrea-all-features-conformance-for-pr",
  "url": "http://10.176.27.169:8080/job/antrea-all-features-conformance-for-pr/",
  "color": "blue"
}
{
  "_class": "hudson.model.FreeStyleProject",
  "name": "antrea-all-features-conformance-for-pull-request",
  "url": "http://10.176.27.169:8080/job/antrea-all-features-conformance-for-pull-request/",
  "color": "disabled"
}
{
  "_class": "hudson.model.FreeStyleProject",
  "name": "antrea-conformance-for-pull-request",
  "url": "http://10.176.27.169:8080/job/antrea-conformance-for-pull-request/",
  "color": "blue"
}
{
  "_class": "hudson.model.FreeStyleProject",
  "name": "antrea-e2e-for-pull-request",
  "url": "http://10.176.27.169:8080/job/antrea-e2e-for-pull-request/",
  "color": "blue"
}
{
  "_class": "hudson.model.FreeStyleProject",
  "name": "antrea-networkpolicy-for-pull-request",
  "url": "http://10.176.27.169:8080/job/antrea-networkpolicy-for-pull-request/",
  "color": "red"
}
{
  "_class": "hudson.model.FreeStyleProject",
  "name": "antrea-tce-integration-for-pull-request",
  "url": "http://10.176.27.169:8080/job/antrea-tce-integration-for-pull-request/",
  "color": "notbuilt"
}
{
  "_class": "hudson.model.FreeStyleProject",
  "name": "antrea-whole-conformance-for-pull-request",
  "url": "http://10.176.27.169:8080/job/antrea-whole-conformance-for-pull-request/",
  "color": "blue_anime"
}
{
  "_class": "hudson.model.FreeStyleProject",
  "name": "theia-daily-flow-visibility-validate-for-period",
  "url": "http://10.176.27.169:8080/job/theia-daily-flow-visibility-validate-for-period/",
  "color": "red"
}
{
  "_class": "hudson.model.FreeStyleProject",
  "name": "theia-e2e-for-pull-request",
  "url": "http://10.176.27.169:8080/job/theia-e2e-for-pull-request/",
  "color": "blue"
}
```

Reply:
Thanks for the details. I think adding a brief comment in code is enough.

Reply:
done.